public/client.js

```diff
@@ -0,0 +1,133 @@
+var ViewControlsDisplay =  Backbone.View.extend({/*properties*/});
+var viewControlsDisplay = new ViewControlsDisplay({el: '#controlsdisplay'});
+
+var ViewListDisplay =  Backbone.View.extend({/*properties*/});
+var viewListDisplay = new ViewListDisplay({el: '#catalogue'});
+
+var ViewAddDisplay = Backbone.View.extend({/*properties*/});
+var viewAddDisplay = new ViewAddDisplay({el: '#adddisplay'});
+
+var RcRecord = new Backbone.Model.extend({
```

Comment:
ok I’m fixing this up seems to be getting better. I’m committing regularly

From: jamdon 
Sent: Friday, June 17, 2016 4:24 PM
To: JSCRIPT300-Spring2016/Final_Project 
Cc: seancava ; Author 
Subject: Re: [JSCRIPT300-Spring2016/Final_Project] Final Project Proposal (#10)

In public/client.js:

> @@ -0,0 +1,133 @@
> +var ViewControlsDisplay =  Backbone.View.extend({/*properties*/});
> +var viewControlsDisplay = new ViewControlsDisplay({el: '#controlsdisplay'});
> +
> +var ViewListDisplay =  Backbone.View.extend({/*properties*/});
> +var viewListDisplay = new ViewListDisplay({el: '#catalogue'});
> +
> +var ViewAddDisplay = Backbone.View.extend({/*properties*/});
> +var viewAddDisplay = new ViewAddDisplay({el: '#adddisplay'});
> +
> +var RcRecord = new Backbone.Model.extend({
When you call Backbone.Model.extend() it returns a constructor function. Don't use new here, as that will return an instance - then, when you try to call new on the instance right after this, it will generate an error. So the steps are:

  1.. define your new constructor by calling Backbone.Model.extend(); - assign this to a constructor variable (i.e. a capitalized variable such as MyModel) 
  2.. create a new instance of that component by calling new MyModel(), assigning it to an instance variable, such as myModel. 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.